presto-orc/src/main/java/com/facebook/presto/orc/TupleDomainFilterUtils.java

```diff
@@ -131,18 +131,22 @@ public static TupleDomainFilter toFilter(Domain domain)
             }
         }
 
-        return MultiRange.of(rangeFilters, nullAllowed, isNanAllowed(ranges));
+        if (firstRangeFilter instanceof DoubleRange || firstRangeFilter instanceof FloatRange) {
+            // != and NOT IN filters should return true when applied to NaN
+            // E.g. NaN != 1.0 as well as NaN NOT IN (1.0, 2.5, 3.6) should return true; otherwise false.
+            boolean nanAllowed = isNotIn(ranges);
+            return MultiRange.of(rangeFilters, nullAllowed, nanAllowed);
+        }
+
+        return MultiRange.of(rangeFilters, nullAllowed, false);
     }
 
     /**
-     * Returns true is ranges represent != or NOT IN filter. These types of filters should
-     * return true when applied to NaN.
-     *
-     * E.g. NaN != 1.0 as well as NaN NOT IN (1.0, 2.5, 3.6) should return true; otherwise false.
+     * Returns true is ranges represent != or NOT IN filter for double, float or string column.
```

Comment:
@kewang1024 string columns are not yet supported here; either add support or update comment

Reply:
Nice catch! Thanks